racket/collects/racket/function.rkt

```diff
@@ -49,6 +50,15 @@
       [(2) (lambda (x y) (not (f x y)))]
       [else (compose1 not f)]))) ; keyworded or more args => just compose
 
+(define (get-failure-result v)
+  (if (procedure? v)
+      (if (procedure-arity-includes? v 0)
+          (v)
+          (raise-argument-error 'get-failure-result
+                                "a procedure that accepts 0 non-keyword arguments"
```

Comment:
This should use `(procedure-arity-includes/c 0)` as the contract string, I think.

Reply:
The intent here may have been to match the exception raised by `hash-ref`. It needs to different, though, to do that. The `hash-ref` function conceptually always applies a procedure failure result to 0 arguments, but it improves the `exn:fail:contract:arity` error message.

Reply:
The error message is based on the one produced by `failure-result/c`, which is really just the error message produced by `(-> any)`. I’m happy to change it, if desired, but consistency with `failure-result/c` seemed logical.